storage.js

```diff
@@ -1,38 +1,27 @@
-// Retrieve the form data from local storage
-function getFormData() {
-  const formData = localStorage.getItem('formData');
-  return formData ? JSON.parse(formData) : {};
-}
+const formD = document.querySelector('.form');
+const uName = document.querySelector('#fullname');
+const msg = document.querySelector('#message');
+const emailD = document.querySelector('#email');
 
-// Save the form data to local storage
-function saveFormData(data) {
-  localStorage.setItem('formData', JSON.stringify(data));
-}
-
-// Update the form fields with the saved data
-function updateFormFields() {
-  const formData = getFormData();
-  document.getElementById('fullname').value = formData.fullname || '';
-  document.getElementById('email').value = formData.email || '';
-  document.getElementById('message').value = formData.message || '';
+function getFormData() {
+  const formData = {
+    uName: uName.value,
+    emailD: emailD.value,
+    msg: msg.value,
+  };
+  return formData;
 }
 
-// Save the form data whenever a field is changed
-function handleInputChange(event) {
-  const inputField = event.target;
-  const fieldName = inputField.fullname;
-  const fieldValue = inputField.value;
-
+formD.addEventListener('submit', () => {
```

Comment:
- [ ] Ouch! this is really close but as mentioned previously and as part of the project requirements the form data should be saved on local storage when the user changes the content of any input field, but I noticed you're saving it upon submission so please make sure to fix this.
- **HINT:** You'll have to use the `change` event for each input field. 😉